designs/configurable-registeration-ttl.md

```diff
@@ -0,0 +1,65 @@
+# RFC: Configurable registeration TTL
+
+## Motivation
+
+[Issue](https://github.com/kubernetes-sigs/karpenter/issues/357)
+
+Karpenter currently hardcodes [registration TTL](https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/nodeclaim/lifecycle/liveness.go#L44) which means nodes must take < 15 minutes to come up and register with the Kubernetes control plane.
+
+This is problematic because some users have nodes that take longer that time to register. As mentioned in the issue, nodes that run GPU can often take > 15 minutes to start up. 
+Use cases:
+- GPU driver installation during node bootstrap
+- Windows AMIs security tooling installation 
+
+## Configuration
+
+### Option 1 (Preferred):
+#### Introduce a configuration registeration TTL to Node Pool then propagate to node claim
+
+We propose introducing a new argument in the Node Pool and propagating it to the NodeClaim during its creation. When the liveness controller evaluates the registrationTTL, it will use the value specified in the NodeClaim.
+
+Importantly, even if the registrationTTL is updated on the NodePool during NodeClaim registration, we will continue to use the NodeClaim's registrationTTL. This approach ensures clarity by demonstrating that the NodeClaim relies on variables defined in its own spec.
+
+Furthermore, this variable will not be considered when evaluating drift, as it is only utilized during the registration process. If a node requires a longer TTL, it would have already been terminated by Karpenter. In cases where the node successfully registers, the TTL becomes irrelevant. Consequently, changes to this variable do not necessitate updates to the drift evaluation process.
+
+Lastly, we will move registrationTTL to fully be in NodePool CRD and default it to 15m.
+
+```yaml
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  annotations:
+    karpenter.sh/nodepool-hash: "13817533045586861872"
+    karpenter.sh/nodepool-hash-version: v3
+  creationTimestamp: "2025-02-12T17:59:44Z"
+  name: example-nodepool
+spec:
+  disruption: {}
+    consolidateAfter: 5m
+    consolidationPolicy: WhenEmptyOrUnderutilized
+  limits:
+    nodes: "800"
+  template:
+    metadata: {}
```

Comment:
nit: maybe a nodepool stub is enough to showcase the actual field of interest. We don't need to show the annotations, node limits etc.

Reply:
cleaned up